Automattic/jetpack
# Boost: Refactor module deactivate and uninstall methodology

Fixes #20760 

#### Changes proposed in this Pull Request:
* Add deactivate and uninstall actions.
* Update base module to hook into these actions and run respective methods which are overridable by individual module.
* Move Critical CSS module deactivation tasks to overridden uninstall tasks within the module.

#### Jetpack product discussion
Slack: C0232B8CUHG/p1631772307009100

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

1. Go to a site with Jetpack Boost Plugin activated.
2. Navigate to wp-admin/admin.php?page=jetpack-boost.
3. Currently there are no deactivation tasks in any individual modules. So, create a temporary `on_deactivate` method under `app/modules/critical-css/class-critical-css.php` and add an error log.
4. Navigate to /wp-admin/plugins.php
5. Deactivate the plugin `Jetpack Boost`
6. Verify that the error is logged
7. Repeat the process for Uninstall. For this use the `on_uninstall` method in step 3, and delete the plugin in step 5. (Delete the plugin in a different test environment, not the one with your main git repository to not accidentally delete a part of the repository.)
